test/utils_test.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 Google Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:pageloader/utils.dart';
+import 'package:test/test.dart';
+
+// Test failures only; successes are tested in matchers_test.dart
+void main() {
+  final utilsError = throwsA(new isInstanceOf<PageLoaderUtilsError>());
+  final foo = 'foo';
+
+  test('exists fails', () {
+    expect(() => exists(foo), utilsError);
+  });
+
+  test('notExists fails', () {
+    expect(() => notExists(foo), utilsError);
```

Comment:
Oh, I hadn't noticed the `notExists` matcher before.

The matcher name `notExists` doesn't follow our Dart guidelines:

- https://www.dartlang.org/guides/language/effective-dart/design#prefer-a-non-imperative-verb-phrase-for-a-boolean-property-or-variable
- https://www.dartlang.org/guides/language/effective-dart/design#prefer-the-positive-name-for-a-boolean-property-or-variable

@munificent @kwalrath: do you have a better name to suggest, or is dropping this matcher (given that we have `exists`) the better way to go?

Reply:
I'm personally in favor of keeping `notExists` and we do use this internally without anyone having any issues with the naming.

My argument would be that this is not a boolean property or variable, but a function instead. So it follows mostly https://www.dartlang.org/guides/language/effective-dart/design#prefer-a-noun-phrase-or-non-imperative-verb-phrase-for-a-function-or-method-if-returning-a-value-is-its-primary-purpose

Reply:
Ok